### ISSUE TITLE
[BUG/Help] <title> The dtype of attention mask (torch.int64) is not bool

### ISSUE THREAD

**Issue by wqh17101** [THUDM/ChatGLM-6B#521]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

There is a warning /ERROR `The dtype of attention mask (torch.int64) is not bool` for  https://huggingface.co/THUDM/chatglm-6b/commits/main   commit id: cde457b39fe0670b10dd293909aab17387ea2c80

### Expected Behavior
No error or warning
It seems that https://huggingface.co/THUDM/chatglm-6b/blob/aa51e62ddc9c9f334858b0af44cf59b05c70148a/modeling_chatglm.py#L306 
Here should be 
```
 attention_scores.masked_fill_(attention_mask > 0, -10000.0)
```

### Steps To Reproduce

model.chat

### Environment

```markdown
- OS:Linux
- Python:3.9
- Transformers:4.27.4
- PyTorch:1.10.1 /2.0.0
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :True
```

**Comment by duzx16** [comment]
This is a warning that can be ignored. The `attention_mask` of dtype `int64` is passed by the `generation` method of `transformers` and not needed for our model.
